test/python/test_variableops_static_input.py

```diff
@@ -0,0 +1,153 @@
+import json
+
+import ngraph_bridge
+import os
+
+# If the below graph is run for many iterations
+# The NGraphVar's NGTensor is updated every iteration
+# NGraphVar's TFTensor is not updated as no TF Node needs it
+# However StaticInputs are derived from the input TF Tensor (which is stale)
+# giving functionally incorrect results
+# TF MeanOp expects a static input
+#
+#    Const     NGraphVar     Const
+#      \       /   |   \     /
+#      _\|  *|/_   |   _\| |/_
+#         Mean     |    Add
+#                  |     /
+#                 \|/  |/_
+#                NGraphAssign
+#
+# After Encapsulation
+#
+#            NGraphVar
+#             /   |   \
+#          *|/_   |   _\|
+#     NGEncap1    |   NGEncap2
+#                 |     /
+#                \|/  |/_
+#             NGraphAssign
+#
+
+from common import NgraphTest
+
+
+class TestVariableStaticInputs(NgraphTest):
+
+    def test_variable_static_input(self):
+
+        def run_test(sess):
+            # Var is initialized by var_init
+            var = tf.get_variable('var', [1], dtype=tf.int32)
+            var_init = tf.constant([0])
+            var_initialize = var.assign(var_init)
+
+            # Computation of mean
+            input1 = tf.constant(
+                [[[1.0, 2.0], [3.0, 4.0]], [[1.0, 2.0], [3.0, 4.0]]],
+                name='input1')
+            mean = tf.reduce_mean(input1, var)
+
+            # For updating the Var
+            const_var = tf.constant([1])
+            var_add = tf.add(var, const_var)
+            var_update = var.assign(var_add)
+
+            # update to happen after mean computation
+            with tf.control_dependencies([mean]):
+                var_update = var.assign(var_add)
+
+            with tf.control_dependencies([var_update]):
+                update_op = tf.no_op('train_op')
+
+            # Initialize Var
+            var_init_value = sess.run((var_initialize))
+
+            # Compute mean and var updates
+            mean_values = []
+            for i in range(3):
+                (result_mean, result_up) = sess.run((mean, update_op))
+                mean_values.append(result_mean)
+
+            # Compute Final Values
+            var_final_val = var.eval(sess)
+            return var_init_value, mean_values, var_final_val
+
+        # This test is not applicable for CPU as NGVariable's NG and TF Tensors
+        # share buffer on CPU. To simulate other backend's non buffer sharing
+        # property we can use this env flag NGRAPH_TF_NGVARIABLE_BUFFER_SHARING
+
+        # set env variable to disable NGraphVariable's buffer sharing
+        buffer_sharing_env = "NGRAPH_TF_NGVARIABLE_BUFFER_SHARING"
+        is_buffer_sharing_env_set = self.is_env_variable_set(buffer_sharing_env)
+        if is_buffer_sharing_env_set:
+            buffer_sharing_env_val = self.get_env_variable(buffer_sharing_env)
+            self.unset_env_variable(buffer_sharing_env)
+
+        self.set_env_variable(buffer_sharing_env, "0")
```

Comment:
We can also repeat the test for `NGRAPH_TF_NGVARIABLE_BUFFER_SHARING` set to 1 and not set. Or maybe another test where `NGRAPH_TF_NGVARIABLE_BUFFER_SHARING` is tested somehow?

Reply:
NGRAPH_TF_NGVARIABLE_BUFFER_SHARING setting to 1 or not setting at all, will show the same behavior on CPU. 
I can unset it and compare the values

Reply:
Added a test with NGRAPH_TF_NGVARIABLE_BUFFER_SHARING=1 set